### DISCUSSION TITLE
Model viewer as 360 panorama

### DISCUSSION THREAD

**Discussion by Nickonimus** [google/model-viewer#3953] [Q&A]
Hi

I would like to know, is it possible to use model viewer as panoramic 360 image viewer?
First reason, i can't set autorotation in other viewers.
Second reason, it would look more consistent and minimalistic to use same viewer for both, 3d models and images. 

Seems like it doesn't work without source glb model, so is there some workaround, and does it have any sense?

**Comment by danilokrulj** [comment]
Maybe make a placeholder model (box or something so it uses as less memory as possible), and set model visibility to none? Since it wont work without source glb model.

**Comment by elalish** [comment]
Yes, setting your image as 'skybox-image' and src to some empty GLB (or an arbitrary GLB and set camera-target far away from it) should work. You'll probably want `orbit-sensitivity="-1"` as well. Maybe a larger field-of-view too.

**Comment by Nickonimus** [comment]
Thanks for answers!
Works exactly like i wanted.
But a problem appeared with colors. It looks like there applied some filmic-alike color managment intended to normalize HDR images, so LDR images unfortunately look pale.

**Reply by elalish** [reply]
Can you post some comparison images to help get to the bottom of this? I thought we had LDR skyboxes working pretty well: https://modelviewer.dev/examples/lightingandenv/#ldrEnvironmentImage, but there's been a long-standing bug with the example below that one. Perhaps we can finally figure this out.

**Reply by Nickonimus** [reply]
Upper image .webp (same for jpg and tif), bright areas slightly grayed out, and shadows a bit more contrast so i can see lose of details.
Bottom image from A-frame viewer. 
![Screenshot 2022-11-23 130731](https://user-images.githubusercontent.com/118725251/203584753-c86026cf-8577-4cce-9aab-bf6459d3dcf8.png)

**Reply by elalish** [reply]
very helpful, thanks! Can you also include the equirect so I can repro?

**Reply by Nickonimus** [reply]
Yes, of course.
Sorry for the long replies, here in ukraine no electricity and internet because endless russian missile attacks. 

Link to pano: https://storage.googleapis.com/empwr_site_files/pano_draft/pano_draft_5k_q90.webp
You can access it directly from localhost:8000

**Reply by elalish** [reply]
I am deeply impressed with your correct diagnosis of such a subtle error! The problem is upstream in three.js and I just made a PR: https://github.com/mrdoob/three.js/pull/25134. You'll also probably be interested in a bug I just fixed regarding empty GLBs: https://github.com/google/model-viewer/issues/3999

I'm so sorry for what you're going through - my best to you and your countrymen.

**Reply by Nickonimus** [reply]
Great, thanks for your time!
Just first thing that came on my mind, since tone mapping widely used in CGI industry :)

Thanks again!

**Reply by elalish** [reply]
Should be fixed now with #4018.
